File: Java/Niveau1/8- Répétitions conditionnées/Département de chimie.java
import algorea.Scanner;

class Main{
    static Scanner entrée = new Scanner(System.in);     

    public static void main(String[] args){
        int nombreMesures = entrée.nextInt();
        int temperatureMinimum = entrée.nextInt();
        int temperatureMaximum = entrée.nextInt();
        boolean safe = true;
        int tour = 1;
        while((tour <= nombreMesures)){
            int mesure = entrée.nextInt();
            if((temperatureMinimum <= mesure) && (mesure <= temperatureMaximum)){
                print("Rien à signaler");
            }else{
                print("Alerte !!");
                safe = false;
                break;
            }
            tour++;
        }
    }

    private static void print(Object object){
        System.out.println(object);
    }
}

File: C++/Niveau 1/8 - Répétitions conditionnées/Département de chimie.cpp
#include <iostream>
using namespace std;

int main(){
    int nombreMesures, temperatureMin, temperatureMax;

    cin >> nombreMesures >> temperatureMin >> temperatureMax;

    int i = 0;

    while (i < nombreMesures){
        int temperature;

        cin >> temperature;

        if((temperature < temperatureMin) || (temperature > temperatureMax)){
            cout << "Alerte !!" << endl;
            break;
        }else{
            cout << "Rien à signaler" << endl;
        }

        i++;

    }

}